api/internal/localizer/localizer.go

```diff
@@ -75,9 +76,69 @@ func (lc *Localizer) Localize() error {
 	return nil
 }
 
-// TODO(annasong): implement
-// processKust returns a copy of the kustomization at kt with paths localized.
-func (lc *Localizer) processKust(kt *target.KustTarget) *types.Kustomization {
+// processKust returns a copy of the kustomization at kt with the patches field localized
+func (lc *Localizer) processKust(kt *target.KustTarget) (*types.Kustomization, error) {
 	kust := kt.Kustomization()
-	return &kust
+	for name, patches := range map[string][]types.Patch{
+		"patches":         kust.Patches,
+		"patchesJson6902": kust.PatchesJson6902,
+	} {
+		for i := range patches {
+			if patches[i].Path != "" {
+				newPath, err := lc.localizeFile(patches[i].Path)
+				if err != nil {
+					return nil, errors.WrapPrefixf(err, "unable to localize %s path", name)
+				}
+				patches[i].Path = newPath
+			}
+		}
+	}
+	return &kust, nil
+}
+
+// localizeFile localizes file path and returns the localized path
+func (lc *Localizer) localizeFile(path string) (string, error) {
+	content, err := lc.ldr.Load(path)
+	if err != nil {
+		return "", errors.Wrap(err)
+	}
+
+	var locPath string
+	if loader.IsRemoteFile(path) {
+		if !lc.addLocalizeDir() {
+			return "", errors.Errorf("cannot localize remote %q: localize directory already exists at %q", path,
+				lc.ldr.Root())
+		}
+		locPath = locFilePath(path)
+	} else { // path must be relative; subject to change in beta
+		// avoid symlinks; only write file corresponding to actual location in root
+		// avoid path that Load() shows to be in root, but may traverse outside
+		// temporarily; for example, ../root/config; problematic for rename and
+		// relocation
+		locPath = cleanFilePath(lc.fSys, filesys.ConfirmedDir(lc.ldr.Root()), path)
+		if !lc.guardLocalizeDir(locPath) {
+			abs := filepath.Join(lc.ldr.Root(), locPath)
+			return "", errors.Errorf("invalid local file path %q at %q: localize directory already exists", path, abs)
+		}
+	}
+	cleanPath := lc.dst.Join(locPath)
```

Comment:
do you mind adding a comment explaining why we need cleanPath here or what it is.

Reply:
Does it help that I renamed it to `absPath`? It's the cleaned, absolute path to the file location specified by the `path` argument. The variable is used to both make the directories holding and write the file to the `localize` destination directory.